File: system/utest/perftest/clock-test.cpp
// Copyright 2018 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <perftest/perftest.h>
#include <zircon/syscalls.h>

namespace {

// Performance test for zx_clock_get(ZX_CLOCK_MONOTONIC).  This is worth
// testing because it is a very commonly called syscall.  The kernel's
// implementation of the syscall is non-trivial and can be rather slow on
// some machines/VMs.
bool ClockGetMonotonicTest() {
    zx_clock_get(ZX_CLOCK_MONOTONIC);
    return true;
}

bool ClockGetUtcTest() {
    zx_clock_get(ZX_CLOCK_UTC);
    return true;
}

bool ClockGetThreadTest() {
    zx_clock_get(ZX_CLOCK_THREAD);
    return true;
}

bool TicksGetTest() {
    zx_ticks_get();
    return true;
}

void RegisterTests() {
    perftest::RegisterSimpleTest<ClockGetMonotonicTest>("ClockGetMonotonic");
    perftest::RegisterSimpleTest<ClockGetUtcTest>("ClockGetUtc");
    perftest::RegisterSimpleTest<ClockGetThreadTest>("ClockGetThread");
    perftest::RegisterSimpleTest<TicksGetTest>("TicksGet");
}
PERFTEST_CTOR(RegisterTests);

}  // namespace

File: sdk/dart/zircon/test/clock_test.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// TODO(https://fxbug.dev/84961): Fix null safety and remove this language version.
// @dart=2.9

import 'package:test/test.dart';
import 'package:zircon/zircon.dart';

void main() {
  test('monotonic clock', () {
    final int time0 = System.clockGetMonotonic();
    final int time1 = System.clockGetMonotonic();
    expect(time1, greaterThan(time0));
  });
}
